app/src/main/java/ch/epfl/sdp/blindwar/user/AppStatistics.kt

```diff
@@ -11,89 +17,180 @@ class AppStatistics {
-        lossPercent = b
+        elo = if (opponentElo == elo) {
+                equalElo(result)
+            } else if (opponentElo > elo) {
+                smallerElo(result, opponentElo)
+            } else {
+                greaterElo(result, opponentElo)
+            }
+
     }
 
+    /**
+     * Calculates ELO in case of ELO equality
+     *
+     * @param result the result of the game
+     * @return the new elo of the user
+     */
+    private fun equalElo(result: Result): Int {
+        if (result == Result.LOSS) {
+            return if (elo < standardEqualValue) {
+                0
+            } else {
+                elo - standardEqualValue
+            }
+        } else if (result == Result.WIN) {
+            return elo + standardEqualValue
+        }
+        return elo
+    }
 
-    // function for updating elo in case of win (very simple version)
-    fun eloUpdateWin(opponentElo: Int) {
-        elo += when {
-            opponentElo > elo -> 15
-            opponentElo == elo -> 10
-            else -> 5
+    /**
+     * Calculates ELO in case of elo is smaller than the opponent's
+     *
+     * @param result the result of the game
+     * @return the new elo of the user
+     */
+    private fun smallerElo(result: Result, opponentElo: Int): Int {
+        var ratio = 0 - (opponentElo.toFloat()/ elo.toFloat())
+        if (result == Result.LOSS) {
+            ratio = 0 - ((1/ratio).pow(3))
+        } else if (result == Result.WIN) {
+            ratio = 0 - (ratio).pow(2)
+        }
+        val diff = round((ratio * standardEqualValue)).toInt()
+        return if (elo - diff <= 0) {
+            0
+        } else {
+            elo - diff
         }
     }
 
-    // function for updating elo in case of loss (very simple version)
-    fun eloUpdateLoss(opponentElo: Int) {
-        elo -= when {
-            opponentElo > elo -> 5
-            opponentElo == elo -> 10
-            else -> 15
+    /**
+     * Calculates ELO in case of elo is greater than the opponent's
+     *
+     * @param result the result of the game
+     * @return the new elo of the user
+     */
+    private fun greaterElo(result: Result, opponentElo: Int): Int {
+        var ratio = (elo.toFloat() / opponentElo.toFloat())
+        if (result == Result.LOSS) {
+            ratio = ratio.pow(2)
+        } else if (result == Result.WIN)
+            ratio = 0 - ((1/ratio).pow(3))
+        val diff = round((ratio * standardEqualValue)).toInt()
+        return if (elo - diff < 0) {
+            0
+        } else {
+            elo - diff
         }
     }
 
+
     override fun toString(): String {
-        return "hello" +
-                //correctArray.toString() + wrongArray.toString() + correctPercent.toString() +
-                //wrongPercent.toString() +
-                wins.toString() + losses.toString() +
-                winPercent.toString() + lossPercent.toString() + elo.toString()
+        return "hello " + elo
```

Comment:
Hi!
why 'hello ' is added to elo? elo squared has a special meaning? haha

Reply:
Someone added a tostring that doesn't make much sense, especially with the arrays, so I changed that, but eventually we might just take it out